lib/spack/spack/environment.py

```diff
@@ -1768,7 +1767,8 @@ def _to_lockfile_dict(self):
                 dag_hash_all = s.build_hash()
                 if dag_hash_all not in concrete_specs:
                     spec_dict = s.to_node_dict(hash=ht.build_hash)
-                    spec_dict[s.name]['hash'] = s.dag_hash()
+                    # Assumes new format, since this was just created.
```

Comment:
Without the context of this PR, this comment doesn't say much. Maybe 
```suggestion
                    # Assumes specfile in current format, since this was just created.
```

Reply:
Changed to "Assumes no legacy formats..."